### ISSUE TITLE
If the CLI args are invalid, the return code is not an error

### ISSUE THREAD

**Issue by meganwalker-ibm** [raineorshine/npm-check-updates#1453]
- X] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)

---

## Steps to Reproduce
Call npm-check-updates with an invalid reject option. I've so far only managed to recreate this with a CLI argument of the following format, discovered after some automation went haywire. I'm unsure how wide this is or what specifically about this set of arguments is causing the blowup

Steps:

```
% npx ncu --version
17.1.1
% npx ncu --reject '[' ''\''invalid'\''' ','\''packagelist'\'']'
Checking /Users/megan/Documents/test/package.json
TypeError: Expected pattern to be a non-empty string
    at Ge (/Users/megan/.nvm/versions/node/v18.20.4/lib/node_modules/npm-check-updates/build/index.js:499:21418)
    at Yx (/Users/megan/.nvm/versions/node/v18.20.4/lib/node_modules/npm-check-updates/build/index.js:499:23462)
    at n (/Users/megan/.nvm/versions/node/v18.20.4/lib/node_modules/npm-check-updates/build/index.js:499:23708)
    at /Users/megan/.nvm/versions/node/v18.20.4/lib/node_modules/npm-check-updates/build/index.js:499:2395
    at Array.some (<anonymous>)
    at e (/Users/megan/.nvm/versions/node/v18.20.4/lib/node_modules/npm-check-updates/build/index.js:499:23794)
    at /Users/megan/.nvm/versions/node/v18.20.4/lib/node_modules/npm-check-updates/build/index.js:499:23861
    at Array.some (<anonymous>)
    at e (/Users/megan/.nvm/versions/node/v18.20.4/lib/node_modules/npm-check-updates/build/index.js:499:23848)
    at /Users/megan/.nvm/versions/node/v18.20.4/lib/node_modules/npm-check-updates/build/index.js:499:2321
% echo $?
0
```

## Current Behavior
The return code indicates success.

## Expected Behavior
The return code indicates that a problem occured

Given how npm-check-updates can also use an error return code to indicate packages need upgrading (with `--errorLevel 2`) it might be worth documenting whatever return code this does exit with to mean 'a problem occured' and not 'dependencies can be updated' in that case.

**Comment by raineorshine** [comment]
Hi, thanks for reporting.

Turns out `picomatch` throws an error if passed an invalid pattern, and that wasn't being caught.

Perhaps more crucially, I fixed the exit code so it is non-zero when any unhandled exception occurs.

Published to `v17.1.2`.

> Given how npm-check-updates can also use an error return code to indicate packages need upgrading (with `--errorLevel 2`) it might be worth documenting whatever return code this does exit with to mean 'a problem occured' and not 'dependencies can be updated' in that case.

I only use error codes 0 and 1 to avoid having to maintain separate error code documentation. The error message `Dependencies not up-to-date` should make it clear why it exited.
